Apps/Src/Task_FaultState.c

```diff
@@ -74,15 +74,19 @@ void EnterFaultState() {
             EEPROM_LogError(FAULT_WATCHDOG);
             break;
     }
-    CANData_t data;
-    data.b = 1;
-    CANPayload_t Message;
-    Message.data = data;
-    // Push Trip message to CAN Q
-    CANbus_BlockAndSend(TRIP, Message);
-    // Push Contactor State message to CAN Q
-    data.b = 0;
-    CANbus_BlockAndSend(CONTACTOR_STATE, Message);
+
+    // avoid infinite recursive faults, since our CAN Driver relys on the OS to work
+    if (Fault_BitMap != Fault_OS) {
+        CANData_t data;
+        data.b = 1;
+        CANPayload_t Message;
+        Message.data = data;
+        // Push Trip message to CAN Q
+        CANbus_BlockAndSend(TRIP, Message);
+        // Push Contactor State message to CAN Q
+        data.b = 0;
+        CANbus_BlockAndSend(CONTACTOR_STATE, Message);
+    }
```

Comment:
So if it is an OS Fault, does this mean that we can't send a TRIP message over CAN? Is there something else we should do instead?

Reply:
We can still log the error to the EEPROM.

On other thing I noticed that I didn't get around to adding in this pull request is that there is no field in the Fault_BitMap for if we fault by dropping too many SPI packets in a row. I can open a ticket for this

Reply:
Opened #417 and #418 for the issues mentioned above